coffea/processor/executor.py

```diff
@@ -259,7 +256,19 @@ def _futures_handler(futures, timeout):
             _cancel(futures.pop())
 
 
-def work_queue_executor(items, function, accumulator, **kwargs):
+class ExecutorABC(metaclass=ABCMeta):
+    def __init__(self, **kwargs):
+        self.kwargs = kwargs or {}
```

Comment:
This is a good opportunity to make the executor validate its options, e.g. the derived classes would implement properties with setters to adjust configurations.